sim/rogue/item_sets_phase_8.go

```diff
@@ -13,3 +18,126 @@ func (rogue *Rogue) ApplyFallenRegalityRogueBonus(aura *core.Aura) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Tank 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	rogue.RegisterAura(core.Aura{
+		Label: label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			oldOnStacksChange := rogue.RollingWithThePunchesProcAura.OnStacksChange
+			rogue.RollingWithThePunchesProcAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+				oldOnStacksChange(aura, sim, oldStacks, newStacks)
+				rogue.PseudoStats.DamageDealtMultiplier /= 1 + 0.01*float64(oldStacks)
+				rogue.PseudoStats.DamageDealtMultiplier *= 1 + 0.01*float64(newStacks)
+			}
+		},
+	})
+}
+
+// Your Blade Flurry now also strikes a third target and increases your attack speed by an additional 10%. In addition, each combo point you spend reduces the remaining cooldown on your Blade Flurry by 0.5 sec.
+func (rogue *Rogue) applyScarletEnclaveTank4PBonus() {
+
+	if !rogue.Talents.BladeFlurry {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Tank 4P Bonus"
+
+	if rogue.HasAura(label) {
+		return
+	}
+
+	core.MakePermanent(rogue.RegisterAura(core.Aura{
+		Label: label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			rogue.OnComboPointsSpent(func(sim *core.Simulation, spell *core.Spell, comboPoints int32) {
+				cdReduction := time.Millisecond * time.Duration(500) * time.Duration(comboPoints)
+				rogue.BladeFlurry.CD.ModifyRemainingCooldown(sim, -cdReduction)
+			})
+
+			rogue.bladeFlurryAttackSpeedBonus += 0.1
+			rogue.bladeFlurryTargetCount += 1
+
+		},
+	}))
+}
+
+// Your Rolling with the Punches can now stack up to 10 times, but grants 2% less health per stack. At 10 stacks, each time you Dodge you will gain 15 Energy.
+func (rogue *Rogue) applyScarletEnclaveTank6PBonus() {
+
+	if !rogue.HasRune(proto.RogueRune_RuneRollingWithThePunches) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Tank 6P Bonus"
+
+	if rogue.HasAura(label) {
+		return
+	}
+
+	metrics := rogue.NewEnergyMetrics(core.ActionID{SpellID: 1226956})
+
+	energyProc := rogue.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 1226956},
+		SpellSchool: core.SpellSchoolNature,
+		ApplyEffects: func(sim *core.Simulation, u *core.Unit, spell *core.Spell) {
+			rogue.AddEnergy(sim, 15, metrics)
+		},
+	})
+
+	energyAura := rogue.RegisterAura(core.Aura{
+		ActionID: core.ActionID{SpellID: 1219291},
+		Label:    "S03 - Item - Scarlet Enclave - Rogue - Tank 6P Bonus Energy Gain",
+		OnSpellHitTaken: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if result.Outcome.Matches(core.OutcomeDodge) {
+				energyProc.Cast(sim, result.Target)
+			}
+		},
+	})
+
+	core.MakePermanent(rogue.RegisterAura(core.Aura{
+		Label: label,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			rogue.rollingWithThePunchesBonusHealthStackMultiplier -= 0.02
+			rogue.rollingWithThePunchesMaxStacks += 5
```

Comment:
We can actually do this without using a variable. You should be able to do
```
rogue.RollingWithThePunchesProcAura.MaxStacks += 5
```

Reply:
Yeah I reverted this change because it wasn't working, but adding to the maxStacks property of the aura doesn't seem to be working either when I run my sim. I still only get up to 5 stacks of the aura.

Reply:
Oh, hm ok. I wonder why 🤔 I took a similar approach in the Shamans 6pT4

Reply:
I think it was my browser cache messing something 😭 It works fine now